File: trunk/java/nanocontainer/src/java/org/nanocontainer/integrationkit/ContainerBuilder.java
/*****************************************************************************
 * Copyright (C) NanoContainer Organization. All rights reserved.            *
 * ------------------------------------------------------------------------- *
 * The software in this package is published under the terms of the BSD      *
 * style license a copy of which has been included with this distribution in *
 * the LICENSE.txt file.                                                     *
 *                                                                           *
 *****************************************************************************/
package org.picoextras.integrationkit;

/**
 * @author <a href="mailto:joe@thoughtworks.net">Joe Walnes</a>
 */
public interface ContainerBuilder {

    /**
     * Create, assemble, init and start a new PicoContainer and store it
     * at a given reference.
     *
     * @param containerRef        Where to store the new container.
     * @param parentContainerRef  Parent container (may be null)
     * @param assembler           Strategy for assembling components in container.
     * @param assemblyScope       Argument to be passed to ContainerAssembler.
     */
    void buildContainer(ObjectReference containerRef, ObjectReference parentContainerRef, ContainerAssembler assembler, Object assemblyScope);

    /**
     * Locate a container at the given reference so it can stopped,
     * destroyed and removed.
     *
     * @param containerRef        Where the container is stored.
     */
    void killContainer(ObjectReference containerRef);

}

File: trunk/dotnet/nanocontainer/src/IntegrationKit/ContainerBuilder.cs
using PicoContainer.Defaults;

namespace NanoContainer.IntegrationKit
{
	public interface ContainerBuilder
	{
		/// <summary>
		/// Create, assemble, init and start a new PicoContainer and store it at a given reference.
		/// </summary>
		/// <param name="parentContainerRef">reference to a container that may be used as a parent to the new container (may be null).</param>
		/// <param name="compositionScope">Hint about the scope for composition.</param>
		/// <returns></returns>
		void BuildContainer(IObjectReference contrainerRef, IObjectReference parentContainerRef, object compositionScope);

		/// <summary>
		/// Locate a container at the given reference so it can be stopped, destroyed and removed.
		/// </summary>
		/// <param name="containerRef">Where the container is stored.</param>
		void KillContainer(IObjectReference containerRef);
	}
}